core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonProxyServer.scala

```diff
@@ -44,7 +45,18 @@ private class AmberProducer(
             throw new RuntimeException(s"not supported payload $payload")
 
         }
-        listener.onNext(new Result("ack".getBytes))
+
+        // get little-endian representation of credits
+        var creditVal: Long = 30L // TODO : replace with actual credit value
+        val creditByteArr: Array[Byte] = new Array[Byte](Longs.BYTES)
+        for (x <- 0 until Longs.BYTES - 1) {
+          creditByteArr(x) = (creditVal & 0xff).asInstanceOf[Byte]
+          creditVal >>= Longs.BYTES
+        }
```

Comment:
is there an easier way to convert a val to a byte array?

Reply:
@Yicong-Huang I refactored to use the `java.nio.ByteBuffer` instead, it simplifies things a lot.